qiskit/circuit/library/standard_gates/u1.py

```diff
@@ -332,6 +348,22 @@ class MCU1Gate(ControlledGate):
     This is a diagonal and symmetric gate that induces a
     phase on the state of the target qubit, depending on the state of the control qubits.
 
+    .. warning::
+
+       This gate is deprecated. Instead, the following replacements should be used
+
+       .. math::
+
+           MCU1(\lambda) = MCP(\lambda)
+
+       .. code-block:: python
+
+          circuit = QuantumCircuit(2)
+          circuit.mcp(lambda, 0, 1)
```

Comment:
It would be nice to show a multi-controlled example here 🙂 
```suggestion
          circuit = QuantumCircuit(5)
          circuit.mcp(lambda, list(range(4)), 4)
```